FWCore/ParameterSet/bin/edmWriteConfigs.cpp

```diff
@@ -136,18 +138,34 @@ namespace {
     }
   }
 
+  bool open_temp(std::string& path, std::ofstream& f) {
+    path += "/XXXXXX";
+    std::vector<char> dst_path(path.begin(), path.end());
+    dst_path.push_back('\0');
+
+    int fd = mkstemp(&dst_path[0]);
```

Comment:
```suggestion
    int fd = mkstemp(path.c_str());
```

Reply:
ah, sorry, I see that you need to modify the string... then
```suggestion
    path += '\0';  // ensure that the string is null-terminated
    int fd = mkstemp(path.data());
```

Reply:
C++11 guarantees that the character array inside an `std::string` is NULL terminated.